From a7cae7962bfda90eea67d33dea0710f548d0e348 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Fri, 7 Jun 2019 11:31:21 +0100
Subject: [PATCH 413/587] drm/vc4: Fix T-format modifiers in FKMS.

The wrong vc_image formats were being checked for in the switch
statement. Correct these.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/vc4/vc4_firmware_kms.c b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
index 20db4f7d7705..d0a5e8d16a21 100644
--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -458,10 +458,10 @@ static void vc4_plane_atomic_update(struct drm_plane *plane,
 	switch (fb->modifier) {
 	case DRM_FORMAT_MOD_BROADCOM_VC4_T_TILED:
 		switch (mb->plane.vc_image_type) {
-		case VC_IMAGE_RGBX32:
+		case VC_IMAGE_XRGB8888:
 			mb->plane.vc_image_type = VC_IMAGE_TF_RGBX32;
 			break;
-		case VC_IMAGE_RGBA32:
+		case VC_IMAGE_ARGB8888:
 			mb->plane.vc_image_type = VC_IMAGE_TF_RGBA32;
 			break;
 		case VC_IMAGE_RGB565:
-- 
2.17.1

